src/library_wasmfs_js_file.js

```diff
@@ -37,7 +37,7 @@ mergeInto(LibraryManager.library, {
       },
       read: (file, buffer, length, offset) => {
         var fileData = wasmFS$JSMemoryFiles[file];
-        length = Math.max(0, fileData.length - offset);
+        length = Math.min(length, fileData.length - offset);
```

Comment:
Won't it cause a problem if the second expression is negative?

Reply:
Maybe, something like this?
`Math.min(length, Math.max(0, fileData.length - offset))`

Reply:
Yes, good point, we should check for that error as well. I pushed a fix now to restore it. I did it on a separate line to hopefully make it clearer, which is maybe why I misunderstood the code before...